### PR TITLE
Add SubPath to VolumeMounts

### PR DESCRIPTION
**Pull request by hidalgopl**
Should be merged together with https://github.com/elotl/itzo/pull/106/files

### CODE REVIEW

File: pkg/server/convert.go [hunk 1]
```diff
@@ -204,15 +204,18 @@ func containerToUnit(container v1.Container) api.Unit {
 		unit.VolumeMounts = append(unit.VolumeMounts, api.VolumeMount{
 			Name:      vm.Name,
 			MountPath: vm.MountPath,
+			SubPath:   vm.SubPath,
 		})
 	}
 	unit.VolumeMounts = append(unit.VolumeMounts, api.VolumeMount{
 		Name:      resolvconfVolumeName,
 		MountPath: "/etc/resolv.conf",
+		SubPath: "/etc/resolv.conf",
```
Comment (ldx) [inline pkg/server/convert.go:213]: Why is this needed?

Comment (hidalgopl) [reply pkg/server/convert.go:213]: If we run itzo with podman, we need full host path to create proper mount in podman container. It's used here: https://github.com/elotl/itzo/pull/106/files#diff-1144a5a6e3efa99f25ae495a0c36a0146549aea7f069d72b9d4016d5a49e158cR156

Comment (ldx) [reply pkg/server/convert.go:213]: Since `/etc/resolv.conf` and `/etc/hosts` are of a Kip-specific volume type, PackagePath, that's something that is handled as a special case in itzo. A bind mount is set up using PackagePath.Path in the pod controller, so you will end up with this on a cell:

    root@ip-172-22-2-193:~# ls -l /tmp/itzo/packages/
    total 28
    drwx------ 2 root root 4096 Nov 19 01:07 default-token-rjt7q
    drwx------ 3 root root 4096 Nov 19 01:07 etchosts
    drwx------ 2 root root 4096 Nov 19 01:07 etc-mtab
    drwx------ 3 root root 4096 Nov 19 01:07 kubeconfig
    drwx------ 3 root root 4096 Nov 19 01:07 resolvconf
    drwx------ 2 root root 4096 Nov 19 01:07 rootfs
    drwx------ 2 root root 4096 Nov 19 01:07 shared
    root@ip-172-22-2-193:~# ls -l /tmp/itzo/packages/resolvconf/etc/resolv.conf
    -rw-r--r-- 1 root root 127 Nov 19 01:07 /tmp/itzo/packages/resolvconf/etc/resolv.conf
    root@ip-172-22-2-193:~# ls -l /tmp/itzo/mounts/
    total 4
    lrwxrwxrwx 1 root root   38 Nov 19 01:07 default-token-rjt7q -> /tmp/itzo/packages/default-token-rjt7q
    lrwxrwxrwx 1 root root   37 Nov 19 01:07 etchosts -> /tmp/itzo/packages/etchosts/etc/hosts
    lrwxrwxrwx 1 root root    9 Nov 19 01:07 etc-mtab -> /etc/mtab
    lrwxrwxrwx 1 root root   45 Nov 19 01:07 resolvconf -> /tmp/itzo/packages/resolvconf/etc/resolv.conf
    lrwxrwxrwx 1 root root    1 Nov 19 01:07 rootfs -> /
    drwxr-xr-x 3 root root 4096 Nov 19 01:07 shared

Maybe you have to add a special case for PackagePath volumes in the podman runtime?

Comment (hidalgopl) [reply pkg/server/convert.go]: :+1: good point, gonna add handling PackagePaths in itzo

**Review by ldx — APPROVED** [review]
